src/stores/search.js

```diff
@@ -0,0 +1,172 @@
+import { defineStore } from 'pinia'
+import { computed, reactive, readonly, toRefs } from '@nuxtjs/composition-api'
+
+import { ALL_MEDIA } from '~/constants/media'
+import { useFilterStore } from '~/stores/filter'
+
+import {
+  filtersToQueryData,
+  queryStringToSearchType,
+} from '~/utils/search-query-transform'
+
+/**
+ * Store information about the search type and search query.
+ * Search query is based on the filter values, but has important differences:
+ * - parameter names are the names that the API expects.
+ * - some parameter names correspond to different filter names for different media types:
+ * `categories` can mean `imageCategories` if the current `searchType` is `IMAGE`, or `audioCategories`
+ * if it is `AUDIO`.
+ * - query parameter values are the values that are used in the API query.
+ */
+export const useSearchStore = defineStore('search', () => {
+  const filterStore = useFilterStore()
+  /**
+   * @type {import('../store/types').SearchState}}
+   */
+  const state = reactive({
+    searchType: ALL_MEDIA,
+    query: {
+      q: '',
+      license_type: '',
+      license: '',
+      categories: '',
+      extension: '',
+      duration: '',
+      aspect_ratio: '',
+      size: '',
+      source: '',
+      searchBy: '',
+      // mature can be either 'true' or ''
+      mature: '',
+    },
+  })
+  const { searchType, query } = toRefs(state)
+
+  /**
+   * @param {import('../store/types').SupportedSearchType} type
+   */
+  const setSearchType = (type) => {
+    state.searchType = type
+    filterStore.setSearchType(type)
+  }
```

Comment:
I wonder if instead of having two public functions for setting the search type, we shouldn't just treat the filterStore search type as the single source of truth and add a computed to be able to reference it in this store? It'd be nice to not expose it in both places.